code/ship/shiphit.cpp

```diff
@@ -1556,8 +1557,9 @@ static void ship_vaporize(ship *shipp)
 	ship_objp = &Objects[shipp->objnum];
 	ship_info* sip = &Ship_info[shipp->ship_info_index];
 
-	// create debris shards
-	create_generic_debris(ship_objp, &ship_objp->pos, (float)sip->generic_debris_spew_num, sip->generic_debris_spew_num * 2.0f, 1.4f, true);
+	// create debris shards if allowed
+	if (!sip->flags[Ship::Info_Flags::No_impact_debris])
+		create_generic_debris(ship_objp, &ship_objp->pos, (float)sip->generic_debris_spew_num, sip->generic_debris_spew_num * 2.0f, 1.4f, true);
```

Comment:
This isn't quite the same situation as an "impact" per se, this is when the ship is destroyed by a large beam weapon. While it's not crazy to have a flag to control that, it seems like this is maybe not what you intended? Something similar happens when a small ship is destroyed as well in `shipfx_blow_up_model()`. If you simply don't want to the ship to produce generic debris on death setting +Generic Debris Spew Num: to 0 *should* suffice.

Reply:
Ah indeed, this addition should not be used here!